%% This is file `dib-template.tex',
%%
%% Copyright 2020 Elsevier Ltd
%%
%% This file is part of the 'Elsarticle Bundle'.
%% ---------------------------------------------
%%
%% It may be distributed under the conditions of the LaTeX Project Public
%% License, either version 1.2 of this license or (at your option) any
%% later version.  The latest version of this license is in
%%    http://www.latex-project.org/lppl.txt
%% and version 1.2 or later is part of all distributions of LaTeX
%% version 1999/12/01 or later.
%%
%% The list of all files belonging to the 'Elsarticle Bundle' is
%% given in the file `manifest.txt'.
%%
%% Template article for Elsevier's document class `elsarticle'
%% with harvard style bibliographic references
%%
%% $Id: dib-template.tex 185 2020-08-07 09:06:08Z rishi $
%%
%% Use the option review to obtain double line spacing
%\documentclass[times,review,preprint]{elsarticle}

%% Use the options `final' to obtain the final layout
%% Use longtitle option to break abstract to multiple pages if overfull.
%% For Review pdf (With double line spacing)
%\documentclass[times,review]{elsarticle}
%% For abstracts longer than one page.
%\documentclass[times,review,longtitle]{elsarticle}
%% For Review pdf without preprint line
%\documentclass[times,review,nopreprintline]{elsarticle}
%% Final pdf
\documentclass[times,final]{elsarticle}
%%
%\documentclass[times,final,longtitle]{elsarticle}
%%

%%
%% Stylefile to load DIB template
\usepackage{dib}
\usepackage{framed,multirow}

%% The amssymb package provides various useful mathematical symbols
\usepackage{amssymb}
\usepackage{latexsym}

%% For line numbers
%\usepackage[switch]{lineno}

% Following three lines are needed for this document.
% If you are not loading colors or url, then these are
% not required.
\usepackage{url}
\usepackage[dvipsnames]{xcolor}
\definecolor{newcolor}{rgb}{.8,.349,.1}


\newcommand{\hh}[1]{{\textcolor{orange}{#1}}}
\newcommand{\svp}[1]{{\textcolor{Fuchsia}{#1}}}
\newcommand{\ac}[1]{{\textcolor{WildStrawberry}{#1}}}
\newcommand{\new}[1]{{\textcolor{JungleGreen}{#1}}}

\definecolor{lightgray}{gray}{0.9}
%%
\usepackage{longtable}
\usepackage[colorlinks]{hyperref}

\usepackage{colortbl}

\journal{Data in Brief}

\begin{document}

\verso{Heike Hofmann \textit{et al.}}

\begin{frontmatter}

\dochead{Data Article}
%The article title must include the word 'data' or 'dataset'.
%Please avoid the use of acronyms and abbreviations where possible.
%For co-submission authors, the title should be unique,
%i.e. not the same as your research paper.
%A maximum of 250 characters is allowed.
%\title{Comparing pairs of Land Engraved Areas on fired Bullets: }%
\title{Comparison Features Data from pairs of Bullet Land Engraved Areas}%
%\tnotetext[tnote1]{This is an example for title footnote coding.}
%Tip: here are a few examples of recent suitable article titles - these are short and clear:
%%Adolescent Rat Social Play: Amygdalar Proteomic and Transcriptomic Data
%%Execution Data Logs of a Supercomputer Workload Over its Extended Lifetime
%%Calgary Preschool Magnetic Resonance Imaging (MRI) Dataset]

%%Authors
\author[1,2]{Heike \snm{Hofmann}\corref{cor1}}
%% First author's email
\ead{hofmann@iastate.edu}
\cortext[cor1]{Corresponding author:
  Tel.: +1-515-294-8948;
  fax: +1-515-294-4040;}
\author[3]{Susan \snm{VanderPlas}}
\author[4]{Eric \snm{Hare}}
%\fntext[fn1]{This is author footnote for second author.}
\author[1,2]{Alicia \snm{Carriquiry}}

%%Affiliations
\address[1]{Department of Statistics, Iowa State University, Ames, Iowa 50011, United States}
\address[2]{Center of Statistics and Applications in Forensic Evidence (CSAFE), Iowa State University, Ames, Iowa 50011, United States}
\address[3]{Department of Statistics, University of Nebraska Lincoln, Lincoln, Nebraska 68503, United States}
\address[4]{OmniAnalytics, Address, City and Postal Code, Country}

%\received{1 May 2013}
%\finalform{10 May 2013}
%\accepted{13 May 2013}
%\availableonline{15 May 2013}
%\communicated{S. Sarkar}


\begin{abstract}
\hh{Land engraved areas (LEAs) on fired bullets form the basis of forensic examinations regarding whether these bullets were fired from the same source (the same barrel).  For algorithmic comparisons of similarity 3d topographic imaging forms the basis of assessing similarity. Here, three parts of the statistical analysis are included: }
(a) descriptive information for each scan\hh{, including its suitability for comparisons}, (b) statistically derived features \hh{characterizing similarity and differences of each pair of topographic LEA scans}, and (c) a fitted Random Forest model object used to \hh{predict the similarity of two scans based on their corresponding similarity features.}
The original raw data is available from the NIST Ballistics Toolmark Database \citep{nist}; here, we provide the code and intermediate data necessary to create the Random Forest model described in \citet{aoas} to make this work  accessible and reproducible. This also allows  future algorithms to be benchmarked against this random forest bullet matching model.
%%%%
% Please Type your abstract here.
%
% \noindent[The Abstract should describe the data collection process, the analysis
% performed, the data, and their reuse potential. It should not provide
% conclusions or interpretive insights. If your article is being
% submitted via another Elsevier journal as a co-submission, please cite
% this research article in the abstract.
%
% \noindent\textbf{Tip:} do not use words such as
% `study', `results', and `conclusions' because a data article should be
% describing your data only.  Min 100 words - Max 500 words.]
%%%%
\end{abstract}

\begin{keyword}
%% Keywords
%[Include 4-8 keywords (or phrases) to facilitate others finding your
%article online.
%\noindent\textbf{Tip:} Try Google Scholar to find which terms are most common in your
%field. In biomedical fields, MeSH terms are a good 'common vocabulary'
%to draw from]
\KWD Land engraved area\sep Forensic Pattern Analysis\sep Topographic Microscopy\sep 3d Imaging\sep Statistics \sep Machine learning
\end{keyword}

\end{frontmatter}

%% For linenumbers
%\linenumbers
% \section*{Data in Brief Article Template}
% %% main text
% \noindent \href{https://www.journals.elsevier.com/data-in-brief/about-data-in-brief/data-in-brief-faq}%
% {\textit{Data in Brief}}
% is an open access journal that publishes data articles.
% Please note:
% \begin{itemize}
% \item A data article is different to a research article, so it is
% important to \textbf{use the template} below to prepare your manuscript for Data
% in Brief.
% \item A data article should \textbf{simply describe data} without providing
% conclusions or interpretive insights.
% \item Before you start writing your data article you should read the
% guidance on
% \href{https://www.journals.elsevier.com/data-in-brief/policies-and-guidelines/what-data-are-suitable-for-data-in-brief}%
% {What Data are Suitable for Data in Brief}.
% \item It is mandatory that Data in Brief authors share their research data:
% \begin{itemize}
% \item If you have \textbf{raw data} (also referred to as primary, source or
% unprocessed data) relating to any charts, graphs or figures in the
% manuscript, these data must be publicly available, either with the data
% article (e.g. as a supplementary file) or hosted on a trusted data
% repository.
%
% \item If you are describing \textbf{secondary data} you are required to provide
% a list of the primary data sources used \underline{and} to make the full secondary
% dataset publicly available, either with the data article (e.g. as a
% supplementary file) or hosted on a trusted data repository.
%
% \item Although we allow supplementary files, it is preferred that
% authors deposit their data in a trusted data repository ($>$70\% of
% Data in Brief authors now do this). See our?
% \href{https://www.elsevier.com/authors/author-resources/research-data/data-base-linking#repositories}%
% {list of supported data repositories}.
%
% \item For data that, for ethical reasons, require access controls a
% mechanism must be provided so that our Editors and reviewers may access
% these data without revealing their identities to authors (more
% information is provided in the template \hyperlink{target1}{below}).
% \end{itemize}
% \end{itemize}
%
% Have you any questions? See a list of frequently asked questions
% \href{https://www.journals.elsevier.com/data-in-brief/about-data-in-brief/data-in-brief-faq}{here},
% or email our Managing Editors:
% \href{mailto:dib-me@elsevier.com}{dib-me@elsevier.com}. This
% step-by-step?
% \href{https://www.journals.elsevier.com/data-in-brief/about-data-in-brief/how-to-submit-your-research-data-article-data-in-brief}%
% {video}?guide will also tell you how to complete the
% template correctly to maximise your chances of acceptance.
%
% \vskip6pt
% \noindent Authors can submit to Data in Brief in two ways:
%
% \begin{enumerate}
% \item[\bf(1)] \textbf{If you are submitting your data article directly to Data in
% Brief, you can now skip the next section and complete the}
% \hyperlink{target2}{\textbf{Data Article template}}.
%
% \item[\bf(2)] \textbf{If you are submitting your data article to Data in Brief via
% another Elsevier journal as a co-submission (i.e. with a Research
% Article), please read the} \hyperlink{target3}{\textbf{Co-submission Instructions}}
% \textbf{on the next page
% before completing the} \hyperlink{target2}{\textbf{Data Article template}}.
%
% \end{enumerate}
%
% \hypertarget{target3}{}
% \section*{Co-submission Instructions}
% A co-submission to~\textit{Data in Brief}~is done at the same time that you
% submit (or resubmit, after revision) a research article to another
% Elsevier journal. For co-submissions you therefore submit your
% \textit{Data in Brief} data article manuscript via the other journal's
% submission system and \underline{not} directly to
% \textit{Data in Brief} itself.
%
% \vskip6pt\noindent
% The other Elsevier journal's Guide for Authors will state if a
% co-submission is offered by that journal, and any revision letter/email
% you receive from a participating journal will contain an offer to
% submit a data article to \textit{Data in Brief}.
%
% \vskip6pt\noindent
% \textbf{To complete a co-submission you will need to zip your
% ~\textit{Data in Brief}~
% manuscript file and all other files relevant to the
% ~\textit{Data in Brief}~
% submission (including any supplementary data files) into a single .zip
% file, and upload this as a "Data in Brief"-labelled item in the other
% journal's submission system when you submit manuscript to that journal.
% The .zip file will then be automatically transferred to
% ~\textit{Data in Brief}~
% when your research article is accepted for publication in the other
% journal, and when published your original research article and data
% article will link to each other on ScienceDirect.}
%
% \vskip6pt\noindent
% \textbf{As~\textit{Data in Brief}~is open access, a moderate article publication charge
% (APC) fee is payable on publication. For more information about the
% APC, please see
% \href{https://www.elsevier.com/journals/data-in-brief/2352-3409/open-access-journal}%
% {here}.}
%
% \vskip6pt\noindent
% \textit{Data in Brief} \underline{requires} that authors share their research data. This can
% be done by submitting it with the data article (e.g. as a supplementary
% file) or by hosting on a trusted data repository (the latter is
% preferred). Failure to do this will delay publication of your
% co-submission.
%
% \vskip6pt\noindent
% \textbf{If you have any questions, please contact:
% \href{mailto:DIB@Elsevier.com}{DIB@Elsevier.com}}
%
% \vskip6pt\noindent
% Please note, authors should not republish the same data presented in
% their original research article in a \textit{Data in Brief} co-submission, as
% this could constitute duplicate publication; however, \textit{Data in Brief}
% welcomes the publication of any data article that fulfils one or more
% of the following criteria:
%
% \checkmark A description of the supplementary data that would
% previously have been hosted as supplementary electronic files alongside
% your original research article.*
%
% \checkmark A description of the full dataset or additional information
% that will aid reuse of the data.
%
% \checkmark A detailed description of the raw data relating to the
% charts, graphs or figures in your companion research article, if making
% these data available will substantially enhance reproducibility and/or
% reanalysis of the data.
%
% \checkmark Any negative datasets or data from intermediate experiments
% related to your research.
%
% \textsf{X} Review articles or supplemental files from a review article
% are not considered original data and are typically unsuitable for Data
% in Brief.
%
% \vskip12pt\noindent
% * If describing supplementary data that you previously planned to
% publish as supplem
% entary electronic files hosted alongside the original
% research article, it is requested that you either\break
% deposit these in a
% repository (preferred) or submit these to \textit{Data in Brief} alongside the
% data article. \textbf{They should not be published as supplementary files with
% your research article in the other journal}.
%
% \clearpage
\hypertarget{target2}{}
\section*{Data Article template}
% \noindent
% Please fill in the template below. All sections are mandatory unless
% otherwise indicated. Please read all instructions in [square brackets]
% carefully and ensure that you delete all instruction text (including
% the questions) from the template before submitting your article.
%
% \vskip6pt\noindent
% Reminder: A data article simply describes data and should not provide
% conclusions or interpretive insights, so \textbf{avoid} using words such as
% `study', `results' and `conclusions'.
%
% \vskip6pt\noindent
% We would welcome feedback on this template and how it might be
% improved. To provide anonymous feedback via a very short survey, please
% click \href{https://forms.office.com/Pages/ResponsePage.aspx?id=P-50kiWUCUGif5-xXBBnXTeXkbO343VFrbpYVBvxdZtUM05UVjIwM0U4WlRKUldCOTNMRUQwOVRHTy4u}%
% {here}.
%
% \vskip6pt\noindent
% {\small\textbf{\textit{Please delete this line and everything above it before submitting your
% article, in addition to anything in [square brackets] below, including
% in the Specifications Table}}\vskip6pt\hrule\vskip12pt}

{\fontsize{7.5pt}{9pt}\selectfont
%%%
\noindent\textbf{Specifications Table}

% Every section of this table is mandatory.
% Please enter information in the right-hand column and remove all the instructions
\begin{longtable}{|p{33mm}|p{94mm}|}
\hline
\endhead
\hline
\endfoot
Subject                & Data Science / Applied Machine Learning\\
\hline
Specific subject area  & Statistical methods in firearms and toolmark evaluation\\
\hline
Type of data           & Table: meta-info.csv\newline
                         Table: hamby-comparisons.csv\newline
                         Model: csafe-rf2.rds\\
%\clearpage
How data were acquired & Table (meta-info.csv): assembled by the authors; consists of a collection of descriptive elements for each of the raw scans and corresponding land engraved areas.  \newline
 Table (hamby-comparisons.csv): features quantifying the similarity between pairs of 3d topographic surface scans detailing striation marks on land engraved areas (LEAs) of fired bullets. Features were extracted using the R software packages x3ptools (heike/x3ptools@164e631) and bulletxtrctr (heike/bulletxtrctr@dfa140a) as outlined in the pipeline by \citet{aoas} and detailed in the accompanying R script. From available meta information ground truth regarding source is added for each comparison. Values are stored in form of a comma separated value file.\newline
                        Model (csafe-rf2.rds): using the features as training data a random forest is fit to predict same source pairs and different source pairs (R package randomForest 4.6-14).\\
\hline
Data format            & Raw \& Analyzed\\
\hline
Parameters for
data\newline
collection             & Hamby sets \citep{hamby} consist of 35 bullets fired from ten consecutively manufactured P85 Ruger barrels. During the firing process micro-imperfections in form of striations are engraved on parts (LEAs) of each bullet. Striation marks are considered to be unique to a barrel. Marks were scanned under 20x magnification by a NanoFocus confocal microscope. Striae from scans of a total of 420 LEAs ($35 \times 2 \times 6$) from two Hamby sets (sets 173 and 252) were considered.
  \\
\hline
Description of
data\newline
collection             & 2d images were rendered from topographic scans. Visual inspection revealed damage on 18 of the 420 LEAs (such as `tank rash`, large amount of pitting, cracks). For the remaining scans, areas of good striations were identified, bullet curvature was removed, and signatures extracted. Pairwise signatures were aligned and features describing similarity of striae extracted. Extracted features include cross-correlation, number and location of coinciding peaks and valleys, height of peaks and valleys,  and consecutively matching striae among others.
                         \\
\hline
Data source location   &  Institution: Iowa State University\newline
                         City/Town/Region: Ames, Iowa\newline
                         Country: United States\newline\newline

                         Primary data sources: \newline
                         x3p files containing topographic surface data from microscopic scans of 35 bullets each from Hamby Set 173 and Hamby Set 252. These scans are available from the
                         National Ballistic and Toolmark Reseach Database at https://tsapps.nist.gov/NRBTD/. In particular, scans for Hamby Set 173 is available from \url{https://tsapps.nist.gov/NRBTD/Studies/Studies/Details/b5fbda3e-d889-4211-be7e-654e19b7cd40?nm=True&mt=1&m=3&sp=1};  and scans for Hamby Set 252 are available from \url{https://tsapps.nist.gov/NRBTD/Studies/Studies/Details/c09aaa86-5d60-4acb-9031-46dad2c0ad32?nm=True&mt=1&m=3&sp=1}. \\
\hline
\hypertarget{target1}
{Data accessibility}   & Primary sources are hosted on a database established by NIST (National Institute of Standards and Technology); data described here, are hosted by the data share service provided by Iowa State, handled through figshare. \newline

                         Repository name: [Name repository]\newline
                         Data identification number: [provide number]\newline
                         Direct URL to data: [e.g. https://www.data.edu.com - please note,\newline
                         this URL should be working at the time of submission]\newline

                         Instructions for accessing these data:\newline\\

%
%                          [\textit{Important: if your data have access controls a mechanism must also be
%                          provided so that our Editors and reviewers may access these data
%                          without revealing their identities to authors, please include
%                          these instructions with your submission. Please contact the Managing
%                          Editors (DIB-ME@DIB.com) if you have any questions.}]\\
\hline
Related
research\newline
article                & E. Hare, H. Hofmann, A. Carriquiry,
                         Automatic matching of bullet land impressions,
                         Annals of Applied Statistics, 11(4), 2017, 2332--2356.
                         DOI: \url{10.1214/17-aoas1080}\newline
\end{longtable}
}
%%%

\section*{Value of the Data}




\begin{itemize}
\itemsep=0pt
\parsep=0pt
\item
These data facilitate the development of probabilistic methods for evaluating and quantifying similarity between pairs of striation mark evidence on fired bullets.

\item
Reproducibility is a critical component of work in forensic statistics, because it enables comparison between newly developed algorithms and established methods. Thus, other researchers in forensic statistics, and possibly forensic examiners who are interested in validating new methods, will benefit from these data being publicly available. Outside of forensic applications, researchers in statistics with violations of independence or researchers in machine learning may find this dataset useful because of its real-world relevance.

\item This data enables researchers to compare the method described in \citep{aoas} to other methods for analysis of bullet land scans, including those previously published, such as in \citet{cmps}, and methods developed in the future. In addition, this data has been used internally for validation of the method for use across other types of ammunition and other models of handguns; thus, experiments which examine the robustness of the model's application on external test sets may produce new insights by leveraging this data. %\svp{XXX It would also be possible to e.g. predict the change in the model's predictions based on updates to include new data sources, but I don't know whether we want to mention that...XXX}
% Forensic handwriting analysis is used to evaluate the source of ransom notes, forgeries, and other such pieces of handwritten evidence. The dataset serves as a sample to build statistical approaches to handwriting analysis. These data, and the methods based upon them, have potential implications in the judicial process, to ensure a transparent, accessible, repeatable, and more objective consideration of handwritten evidence.

\item Open source algorithms supplemented by open data are a critical component of a fairer justice system, equally accessible to both prosecution and defense. Currently, many closed-source or closed-data algorithms are used to supplement the discretion of law enforcment agents and prosecutors, and these algorithms cannot be easily audited by affected parties. By making the source and intermediate data relating to this algorithm available, we are ensuring that the algorithm's details are fully transparent to all parties who may be affected by its use in forensic casework.
\end{itemize}

\section*{Data Description}
%  [Individually describe each data file (i.e. figure 1, figure 2, table
% 1, dataset, raw data, supplementary data, etc.) that are included in
% this article. Please make sure you refer to every data file and provide
% a clear description for each - do not simply list them. No insight,
% interpretation, background or conclusions should be included in this
% section. Please include legends with any tables, figures or graphs.
%
% \noindent\textbf{Tip:} do not forget to describe any supplementary data files.]
\noindent

An early study in the field of firearm examination was executed by \citet{hamby} and consists of over 300 sets of bullets sent out to firearm examiners worldwide.
Each of these sets consists of 20 bullets with known provenance and 15 questioned bullets: two bullets each were fired from ten consecutively manufactured barrels of P-85 Ruger firearms. Barrels are labeled 1 - 10, and the bullets are labelled as bullet 1 and bullet 2 from a particular barrel. %These two bullets were labeled as bullets 1 and 2 from barrels 1 through 10.
The remaining fifteen bullets were each fired from one of the ten barrels (such that at at least one and at most 4 bullets came from the same barrel) and labeled with uppercase letters. Firearms examiners are then asked to match the questioned bullets to the corresponding barrels.

Each barrel has six land engraved areas resulting from firing the gun; together with the Barrel and Bullet identifiers, the land identifier can be used to identify a single digital scan stored in x3p format\citep{x3p}.

These scans have an additional advantage over conventional sets of physical bullets: it is much easier to share the digital data.
The NIST database \citep{nist} contains the raw data for this study as well as many others.
Information about the individual scans is provided in \autoref{meta}; here, each row describes a scan of a single bullet land, its location on disk after the download, whether the corresponding land shows damage unrelated to the firing, and its relationship to Hamby set 173 bullet 1 from the same barrel.

\autoref{features} consists of a description of all features describing pairwise comparisons of scans. These features are aimed to distinguish between pairs of scans from the same source (corresponding to two bullets fired from the same barrel) and two different sources (corresponding to bullets fired from different barrels).
\autoref{fig:densities} gives an overview of some of the features in the `hamby-comparisons.csv` file. Values from same-source comparisons are shown in orange, different-source comparisons are shown in grey. Area under the curve (AUC) is a value between 0 and 1 describing the strength of one feature's strength in discriminating between same-source and different-source comparisons. The closer values are to 1, the better they discriminate.


\begin{longtable}{p{33mm}p{94mm}}
\caption{\label{meta} Description of the variables in `meta-info.csv`.}
\hline
\bf Table   (meta-info.csv)              \\
\bf Variable name & \bf Description\\ \hline
\endhead
\hline
\endfoot
\hline
\bf Land identifiers\\
\hline\rowcolor{lightgray}
\bf land\_id          & identifier of a land engraved area of the form {\tt study\_barrel\_bullet\_land}. \\
\bf study, barrel, bullet, land & elements describing the provenance of a LEA. \\\rowcolor{lightgray}
\bf source & path to the x3p file corresponding to the LEA. \\
\bf damaged & logical variable indicating whether the land engraved area is exhibiting damage not associated with the barrel and the firing process. \\\rowcolor{lightgray}
\bf H173\_B1\_index & matching land id to bullet 1 of the Hamby 173 set of the barrel the bullet was fired from.  This index enables to determine ground truth in any comparison: two land engraved areas come from the same source if their corresponding {\bf H173\_B1\_index} matches.\\
\hline
\end{longtable}


\begin{longtable}{p{33mm}p{94mm}}
\caption{\label{features} Description of features extracted from pairwise comparisons of scans. These values are stored in file `hamby-comparisons.csv`.}
\hline
\multicolumn{2}{l}{\bf Table   (hamby-comparisons.csv)}             \\
\bf Variable name & \bf Description\\ \hline
\endhead
\hline
\endfoot
\hline
\bf Land identifiers\\
\hline\rowcolor{lightgray}
\bf land1, land2          & identifiers of the land engraved areas of the form {\tt study\_barrel\_bullet\_land} where {\tt study} corresponds to either Hamby173 or Hamby252. A Hamby set consists of twenty bullets of known origin fired from barrels 1 through 10, and 15 questioned bullets. Questioned bullets are named by an uppercase latin letter. They have by definition no known barrel, i.e.\  land 3 from questioned bullet A from set 252 would be denoted as {\tt Hamby252-BrUkn-BA-L3}. Two bullets are fired from each of the ten barrels. Land 2 from the first bullet of barrel 4 in Set 173 is denoted as {\tt Hamby252-Br4-B1-L2}.
{\bf land1} and {\bf land2} uniquely describe one row in the hamby-comparisons.csv data set.
\\
\bf study1, study2, barrel1, barrel2, bullet1, bullet2, l1, l2 & elements describing the provenance of the two LEAs in a comparison. All of these elements can be directly derived from {\bf land1} and {\bf land2} and are added for convenience  only.\\ \hline
\multicolumn{2}{l}{\bf Measures based on comparisons of two aligned signatures} \\ \hline
\bf ccf & maximized cross-correlation between two LEA signatures, \\\rowcolor{lightgray}
\bf rough\_cor & correlation after detrending aligned signatures.\\
\bf length, length\_mm & length of the shorter of two signatures -- as vector length, or converted to millimeter taking resolution into account.\\\rowcolor{lightgray}
\bf lag, lag\_mm, abs\_lag, abs\_lag\_mm & (absolute) lag (in millimeter) between signatures to achieve alignment. \\
\bf overlap & ratio of overlap between the signatures after alignment. The length of the shorter of the two signatures is used as denominator in the ratio. \\\rowcolor{lightgray}
\bf matches, matches\_per\_mm & number of matching peaks (per millimeter) between aligned signatures. Matching in this context is defined as the same (or similar) location along the $x$ axis. \\
\bf mismatches, mismatches\_per\_mm & number of peaks on a signature that does not have a counterpart peak in close proximity along the $x$ axis on the aligned signature. \\\rowcolor{lightgray}
\bf cms, cms\_per\_mm, cms2, cms\_per\_mm, & CMS are \underline{c}onsecutively \underline{m}atching \underline{s}triae \citep{biasotti} between two aligned land engraved areas (LEAs). Striations, as defined by firearms and toolmarks examiners, exhibit as peaks in the  corresponding signature. Because confocal light microscopy also captures depths, we can expand the notion of striations to peaks and valleys. Correspondingly, {\bf cms} is defined as the number of consecutively matching peaks and valleys of two aligned signatures, while {\bf cms2} is the number of just the peaks. \\
\bf non\_cms, non\_cms\_per\_mm & longest sequence of non-matching peaks (per millimeter) between two aligned signatures\\\rowcolor{lightgray}
\bf left\_cms, right\_cms & longest sequence of matching striae from the left/right of two aligned signatures. \\
\bf D, sd\_D & Euclidean distance (in millimeters) between two aligned signatures and its standard deviation. \\\rowcolor{lightgray}
\bf sum\_peaks & the depth of peaks measured as the sum of matching peaks between two aligned signatures (in microns).\\
\bf samesource &  ground truth whether a pair is from the same source (TRUE) or from different sources (FALSE). \\\rowcolor{lightgray}
\bf damaged & logical variable indicating whether one of the land engraved areas shows visible damage such as pitting (holes in the bullet surface from gun powder exploding in the barrel next to it) or "tank rash" (damage to the bullet after it exited the barrel after the firing). \\
\hline
\end{longtable}


<<data-import, echo=FALSE, message=FALSE, warning=FALSE>>=
library(tidyverse)
library(randomForest)
library(here)
features <- read.csv(here("data/hamby-comparisons.csv"))
rf2 <- readRDS(here("data/csafe_rf2.rds"))
important <- data.frame(importance = rf2$importance) %>% arrange(desc(MeanDecreaseGini))
important$name <- row.names(important)
doubles <- duplicated(features$id)
features <- features %>% filter(!doubles, !damaged)

library(pROC)
important$auc <- NA
for (i in 1:nrow(important)) {
  important$auc[i] <- auc(features$samesource, features[,important$name[i]])
}
detach(package:pROC)
@

<<densities, echo=FALSE, message=FALSE, warning=FALSE, fig.width = 10, fig.height=8, out.width='\\textwidth', fig.cap="Densities of same-source and different source comparisons of each  feature used in fitting the random forest model. Features are sorted according to importance (most important features first). AUC (area under the curve) describes each feature's univariate strength to distinguish between same source and different source.">>=

long_rf <- features %>%
  filter(!doubles, land1 != land2) %>%
  select(important$name, samesource) %>%
  pivot_longer(cols = -samesource) %>%
  mutate(
    name = factor(name, levels=important$name),
    `Ground truth` = c("Different source", "Same source")[as.numeric(samesource)+1]
  )


long_rf %>%
  ggplot(aes(x = value)) +
  geom_density(aes(fill=`Ground truth`), alpha = 0.7, size = 0.2) +
  facet_wrap(~factor(name, levels=important$name), scales = "free") +
  scale_fill_manual(values=c("darkgrey", "darkorange")) +
  theme_bw() +
  theme(legend.position = c(.8,0.1)) +
  geom_label(aes(x = Inf, y = Inf, label=sprintf("AUC: %.3f", round(auc,3))), data = important,
            color = "grey50", size = 4.5, vjust=1, hjust=1,label.size = 0) +
  xlab("Value observed in the comparison of two LEA scans")
@

Finally, csafe-rf2.rds is the binary file containing the random forest object fitted on the features in \texttt{hamby-comparisons.csv}. Random forests \citep{breiman} are a machine-learning method using an ensemble of decision trees.
This random forest model is used to assess the similarity of two scans, with an output prediction of 0 indicative of dissimilarity between the scans, and an output prediction near 1 indicative of similarity between the scans. In addition to containing the weights used to make predictions, the fitted model also contains a record of the predictions made during the fitting process; and the features used to fit the model to the original set of scans. It is important to archive not only the code used to fit this model, but also the intermediate data, derived features, and full fitted model object, so that when the model is re-fit we have a record of changes between versions of the random forest models.


The random forest object `csafe\_rf2` has an overall out-of-bag error rate estimated as 0.32\%. Out of the total 81,003 pairwise comparisons of the `hamby-comparisons.csv` data 80,743 comparisons or 99.68\% are correctly classified as same-source or different-source pairs.

\autoref{fig:outcomes} shows a summary of (out-of-bag) prediction values of the csafe\_rf2 random forest trained on the feature values.

<<outcomes, echo=FALSE, message=FALSE, warning=FALSE, fig.align="center", fig.width = 6, fig.height=4, out.width='.7\\textwidth', fig.cap="Summary of the out-of-bag predictions of the csafe\\_rf2 random forest object trained on the features in `hamby-comparisons.csv'.">>=
features$rf_oob <- predict(rf2, type="prob")[,2]

features %>%
  mutate(
    samesource = factor(samesource, labels=c("Different-source", "Same-source")),
    Comparison = samesource
  ) %>%
  ggplot(aes(x = rf_oob, fill = Comparison, colour=Comparison)) +
  geom_density(alpha=0.6) +
  scale_colour_manual(values = c("darkgrey", "darkorange")) +
  scale_fill_manual(values = c("darkgrey", "darkorange")) +
  theme_bw() +
  xlab("Out-of-bag Random Forest Prediction values") +
  facet_grid(samesource~.) +
  theme(legend.position="bottom")
@

\begin{longtable}{p{127mm}}
\hline
\bf Model   (csafe-rf2.rds)              \\ \hline
\endhead
\hline
\endfoot
\hline
Random forest object for assessing similarity of two land engraved areas based on features extracted from aligned signatures. In particular, ccf, rough\_cor,
    matches\_per\_mm, D, sd\_D, cms\_per\_mm, mismatches\_per\_mm,
    overlap, abs\_lag\_mm, sum\_peaks were used to fit the random forest.\\
\hline
\end{longtable}

\section*{Experimental Design, Materials and Methods}

% \noindent [Offer a complete description of the experimental design and methods
% used to acquire these data. Please provide any programs or code files
% used for filtering and analyzing these data. It is very important that
% this section is as comprehensive as possible. If you are submitting via
% another Elsevier journal (a co-submission) you are encouraged to
% provide more detail than in your accompanying research article. There
% is no character limit for this section; however, no insight,
% interpretation, or background should be included in this section.
%
% \noindent\textbf{Tip:} do not describe your data (figures, tables, etc.) in this section,
% do this in the Data Description section above.]

\subsection*{Materials and Data Collection Methods}
A large part of a forensic examiner's job consists of evaluating whether two pieces of evidence come from the same source or from different sources.
In casework, same source can never be established with 100\% certainty; as a result, error rate and proficiency studies are used to establish the validity of forensic evaluation \citep[as recommended in][]{nrc,pcast}.
The random forest model provided in this paper is fitted based on two sets of bullets from a study conducted by \citeauthor{hamby}; the bullets in question were scanned at the National Institutes of Standards and Technology by Xiaoyu Zheng and Wei Chu using a NanoFocus V.6.12.6 confocal light microscope at 20x; the resulting scans have a resolution of 1.5625 $\mu$m per pixel.

The raw scans can be downloaded from \citet{nist} by searching for ``Hamby (2009) Barrel Set" and selecting the two study results matching that name. The studies should be downloaded separately and each placed into a folder.

\newpage

\subsection*{Feature Derivation}
Once the raw data files are downloaded, the code \texttt{create-features.R} can be run to generate the complete set of pairwise comparisons and intermediate results; the feature summaries for each pairwise comparison are, of course, provided in \texttt{hamby-comparisons.csv}. These features are produced using functions from the R package \texttt{bulletxtrctr}; the feature derivation process will be described here.

The initial x3p scans describe a curved surface with striations produced as the bullet is propelled through the barrel during the firing process. Though the source data is three-dimensional, the actual information we use to make the comparison is two-dimensional: it is essentially a cross-section of the scanned surface. The initial processing method is designed to reduce the 3D scan into a two-dimensional "signature" of a bullet land, without the underlying surface curvature. The protocol for the data reduction process is described in great detail in \citet{aoas} and the code provided here leverages \texttt{bulletxtrctr}, an R package designed to implement the process.

Each step in the file \texttt{create-features.R} is described briefly below; all of the code necessary to reproduce the steps exactly is provided in the file. In order to replicate the entire process, you can download the data as directed and run the \texttt{create-features.R} file in the directory where the data is downloaded. The individual functions used are described in detail in the bulletxtrctr R package and the corresponding code can be found there.

\begin{enumerate}
\item The scans are oriented so that the base of the bullet is at the bottom and the bullet scan is curving from left to right. Fired bullets have \emph{breakoff}, where some of the metal flies off of the bullet during firing as a result of heat and pressure. The scans are oriented such that any breakoff should be located in the bottom right corner of the scan. For Hamby 173, bullets were flipped along the $y$ axis; for Hamby 252, the bullets were rotated counterclockwise by 90 degrees and then flipped along the $y$ axis. Finally, scans are converted so that all measurements are in $\mu$m.
\item Once scans are oriented properly, a stable cross-section (one that does not cross through any regions of breakoff) that is as low as possible on the bullet (so as to cross deeper, more well-engraved stria) is identified algorithmically using the cross-correlation between adjacent cross-sections to quantify stability. These cross sections are checked manually using graphics and are set by hand when the algorithmic identification fails. In this dataset, there were no manually identified cross sections.
\item Using the cross-section location identified in step 2, the measurements at that crosscut are extracted from the scan.
\item Damaged scans not suitable for comparison are marked.
\item The grooves that demarcate the land engraved area are identified algorithmically and then manually checked and set as necessary using a Shiny app (code is provided in the file).
\item The areas between the grooves are extracted and a smooth is fit to the data to remove the bullet surface curvature, producing a "bullet signature". This signature is used to represent the entire bullet land engraved area scan.
\item Once signatures have been identified, each scan is compared to every other scan to produce a set of pairwise comparisions. The signatures in each pairwise scan are identified and features are computed using the bulletxtrctr software describing how well the two scans are aligned.
\item The ground truth is included in the feature set for evaluating the prediction algorithm.
\item Duplicate pairwise comparisons are removed so as not to compromise the model fitting process.
\end{enumerate}

\subsection*{Model Fitting}
The random forest model is fit within the statistical computing software R 4.0.1 \citep{R} using the package {\tt randomForest} 4.6-14 \citep{randomforest}. The following lines of code fit the forest object `csafe\_rf2`:

<<fit-forest, echo=TRUE, eval=FALSE, tidy=FALSE>>=
library(randomForest)
# Read in the features
features <- read.csv("data/hamby-comparisons.csv")

set.seed(20140105)
# fit the random forest
csafe_rf2 <- randomForest(
  factor(samesource) ~ ccf + rough_cor + matches_per_mm + D + sd_D +
    cms_per_mm + mismatches_per_mm + overlap + abs_lag_mm + sum_peaks,
  data = features)

saveRDS(csafe_rf2, "data/csafe_rf2.rds")
@


\section*{Ethics Statement}
% \noindent [Please refer to the journal's
% \href{https://www.elsevier.com/journals/data-in-brief/2352-3409/guide-for-authors}{Guide for Authors}
% for more information on
% the ethical requirements for publication in Data in Brief. In addition
% to these requirements:
%\noindent\textbf{If the work involved the use of human subjects:}
%please include a statement here confirming that informed consent was
%obtained for experimentation with human subjects;

% \noindent\textbf{If the work involved animal experiments:} please
% include a statement confirming that all experiments comply with
% the \href{https://www.nc3rs.org.uk/arrive-guidelines}{ARRIVE\ guidelines} and were be carried out in accordance with the
% U.K. Animals (Scientific Procedures) Act, 1986 and associated
% guidelines, \href{https://ec.europa.eu/environment/chemicals/lab_animals/legislation_en.htm}{EU Directive 2010/63/EU for animal experiments}, or the
% National Institutes of Health guide for the care and use of Laboratory
% animals (NIH Publications No. 8023, revised 1978)]


\noindent This study was conducted using publicly available data and open-source code; no human subjects or animal research was involved in this experiment. The authors are not aware of any conflicts of interest.

\section*{Acknowledgments}
% Acknowledgments should be inserted at the end of the paper, before the
% references, not as a footnote to the title. Use the unnumbered
% Acknowledgements Head style for the Acknowledgments heading.

\noindent
This work was funded (or partially funded) by the Center for Statistics and Applications in Forensic Evidence (CSAFE) through Cooperative Agreement 70NANB20H019 between NIST and Iowa State University, which includes activities carried out at Carnegie Mellon University, Duke University, University of California Irvine, University of Virginia, West Virginia University, University of Pennsylvania, Swarthmore College and University of Nebraska, Lincoln.

\section*{Declaration of Competing Interest}
%
% \noindent [All authors are required to report the following information:
% \begin{enumerate}
% \item[(1)] All third-party financial support for the work this article;
%
% \item[(2)] All financial relationships with any entity that could be
% viewed as relevant to data described in this manuscript;
%
% \item[(3)] All sources of revenue with relevance to this work where
% payments have been made to authors, or their institutions on their
% behalf, within the 36 months prior to submission;
%
% \item[(4)] Any other interactions with the sponsor, outside of the
% submitted work;
%
% \item[(5)] Any relevant patents or copyrights (planned, pending or
% issued);
%
% \item[(6)] Any other relationships or affiliations that may be
% perceived by readers to have influenced, or give the appearance of
% potentially influencing, what has been written in this article.
% \end{enumerate}
% As a general guideline, it is usually better to
% disclose a relationship than not. This information will be acknowledged
% at publication in the manuscript. If there is no known competing
% financial interests or personal relationships that could have appeared
% to influence the work reported in this paper, please include this
% statement.]

\noindent
The authors declare that they have no known competing
financial interests or personal relationships which have, or could be
perceived to have, influenced the work reported in this article.

% \vskip12pt\noindent
% [If there are financial interests/personal relationships which may be
% considered as potential competing interests, please declare them here.]

% \subsection*{Note}
% \label{sec1}
% Any instructions relevant to the \verb+elsarticle.cls+ are applicable
% here as well. See the online instruction available on:
% \makeatletter
% \if@twocolumn
% \begin{verbatim}
%  http://support.river-valley.com/wiki/
%  index.php?title=Elsarticle.cls
% \end{verbatim}
% \else
% \begin{verbatim}
%  http://support.river-valley.com/wiki/index.php?title=Elsarticle.cls
% \end{verbatim}
% \fi

%\section*{References}

% \noindent [References are limited (approx. 15) and excessive self-citation is not
% allowed. \textbf{If your data article is co-submitted via another Elsevier
% journal, please cite your associated research article here.}
% \hh{Associated research article is \cite{aoas}.}



%% Numbered
%%If
\bibliographystyle{model1-num-names}
\bibliography{references}

\end{document}

%%
